File: GP/Fase1/G.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n;
    cin >> n;

    vector<int> dp(n+1, 0);
    dp[1] = 1;
    for (int i = 2; i <= n; i++) {
        for (int j = 1; j <= i && j <= 3; j++) {
            if (!dp[i-j]) {
                dp[i] = 1;
                break;
            }
        }
    }

    if (dp[n]) {
        cout << "Jane\n";
    } else {
        cout << "John\n";
    }
    
    return 0;
}

File: GP/Fase1/G.py


n = int(input())

if n == 1:
    print("Jane")
else:
    winner = [None]*(n+1)

    winner[1] = 'Jane'

    winner[2] = 'John'

    for i in range(3, n+1):
        if all(winner[i-j] == 'John' for j in range(1, i)):
            winner[i] = 'Jane'
        else:
            winner[i] = 'John'

    print(winner[n])